pkg/chunk/cache/redis_cache.go

```diff
@@ -50,44 +50,30 @@ func (cfg *RedisConfig) RegisterFlagsWithPrefix(prefix, description string, f *f
 	f.DurationVar(&cfg.MaxConnLifetime, prefix+"redis.max-conn-lifetime", 0, description+"Close connections older than this duration. If the value is zero, then the pool does not close connections based on age.")
 }
 
+// Validate Redis configuration
+func (cfg *RedisConfig) Validate() error {
+	switch cfg.Topology {
+	case redisTopologyServer, redisTopologyCluster, redisTopologySentinel:
+	default:
+		return fmt.Errorf("unsupported Redis topology %q", cfg.Topology)
+	}
+	return nil
+}
+
 // NewRedisCache creates a new RedisCache
-func NewRedisCache(cfg RedisConfig, name string, pool *redis.Pool, logger log.Logger) *RedisCache {
+func NewRedisCache(cfg RedisConfig, name string, logger log.Logger, redisClient RedisClient) *RedisCache {
 	util.WarnExperimentalUse("Redis cache")
 	// pool != nil only in unit tests
```

Comment:
Comment seems obsolete.

I would suggest *always* passing non-nill redisClient to NewRedisCache, and create client outside of this function.